submissions/kotlyar-andrey/oop_exercise/index.js

```diff
@@ -0,0 +1,133 @@
+/* Refer to https://github.com/OleksiyRudenko/a-tiny-JS-world for the task details
+   Complete the below for code reviewers' convenience:
+
+   Code repository: https://github.com/kotlyar-andrey/a-tiny-JS-world
+   Web app: https://kotlyar-andrey.github.io/a-tiny-JS-world/
+   */
+
+/**
+ * Общий класс для всех жителей
+ */
+ class Creature {
+  constructor(species, name, gender, phrase) {
+    this._species = species;
+    this._name = name;
+    this._gender = gender;
+    this._phrase = phrase;
+    this._friends = [];
+    this._properties = ["_species", "_name", "_gender"];
+  }
+
+  getFriends() {
+    return this._friends && this._friends.length > 0
+      ? this._friends.map(friend => friend._name).join(", ")
+      : "no friends";
+  }
+
+  setFriends(newFriends) {
+    this._friends = [...newFriends];
+  }
+
+  getSaying() {
+    return this._phrase;
+  }
+
+  setSaying(newPhrase) {
+    this._phrase = newPhrase;
+  }
+
+  addProperties(...newProperties) {
+    this._properties = [...this._properties, ...newProperties];
+  }
+
+  display() {
+    const friends = this.getFriends();
+    const saying = this.getSaying();
+    print(
```

Comment:
A class shouldn't know anything about external environment.
Using a side effect (`print` is a side effect) also makes the class less flexible.

Reply:
Ok, I can rename display to getInfo and return value
  ```
getInfo() {
    const friends = this.getFriends();
    const saying = this.getSaying();
    return this._properties
      .map(prop => this[prop])
      .join("; ")
      .concat(`; ${saying}; ${friends}`);
  }
```
  End then print inhabitants
 ```
 inhabitants.forEach(creature => {
    print(creature.getInfo());
});
```

Reply:
Making class method just return some data is definitely a good design decision.